### DISCUSSION TITLE
Possible to define a cache-handler.ts file ?

### DISCUSSION THREAD

**Discussion by AlexandreBourdeaudhui** [caching-tools/next-shared-cache#392] [Q&A] [answered]
Hello, I'm using the Cache-Handler package with an ioredis stack. I already have a file where all the logic for connecting to Redis is located. However, I'm unable to reuse this logic in the cache-handler.mjs file because it's a TS file, and Next.js returns an error 'TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts' during my tests. Is there a way to use a .TS file or reuse the Redis connection logic?

**Comment by better-salmon** [comment, marked as answer]
Hey @AlexandreBourdeaudhui!

Unfortunately, Next.js custom cache API expects [a path to a JS file](https://nextjs.org/docs/app/api-reference/next-config-js/incrementalCacheHandlerPath). Internally, they call a constructor without building or transpiling the cache-handler code.

So the answer is no 😢
